api/envoy/config/core/v3/health_check.proto

```diff
@@ -178,6 +179,10 @@ message HealthCheck {
     // payload block must be found, and in the order specified, but not
     // necessarily contiguous.
     repeated Payload receive = 2;
+
+    // When setting this value, it tries to attempt health check request with ProxyProtocol.
+    // Payload is sent after preceding ProxyProtocol header.
+    ProxyProtocolConfig proxy_protocol_config = 3;
```

Comment:
For the comment, would prefer: "Use the proxy protocol for health checks. The send payload is sent immediately after the proxy protocol header."

Furthermore, is it valid to have an empty `send` payload when this flag is enabled? If so we should both document that and guard against configurations that leave `send` empty.

Reply:
Sorry I forcefully pushed. The fixed commits are shown here. https://github.com/envoyproxy/envoy/pull/31691/commits/495ba97c84b7b442fc51d791bdff37ce932cd9a5#diff-4ccbbf045baa1ad3abd43bb7534d765e115e50be2ac9d9c3639588a4fd54e8a6R183-R185